### PR TITLE
Bugfix/correct contains and starts with logic

### PR DESCRIPTION
**Pull request by jestingrabbit**
Mysql docs make it clear that these `match..against` clauses were a bit off.

https://dev.mysql.com/doc/refman/8.0/en/fulltext-boolean.html

### CODE REVIEW

File: spec/winnow/model_spec.rb [hunk 1]
```diff
@@ -159,7 +159,7 @@
           [Struct.new(:columns, :type).new(["name"], :fulltext)]
         end
 
-        expect_any_instance_of(ActiveRecord::Relation).to receive(:where).with(fts_scope, "test*", "%test%")
+        expect_any_instance_of(ActiveRecord::Relation).to receive(:where).with(fts_scope, "", "%test%")
```
Comment (jestingrabbit) [inline spec/winnow/model_spec.rb:162]: Not sure if the blank here is gonna be okay ie does it throw some huge inefficiency into the query?

File: lib/winnow/model.rb [hunk 2]
```diff
@@ -83,12 +83,20 @@ def fts_scope_for(column)
...
       end
 
-      def fts_tokens_for(term)
+      SPECIAL_CHARS = %r{[@~"<>{}()+*\-]+}
+
+      def fts_starts_with_tokens_for(term)
         # since we're searching in boolean mode, strip out search operators and tokenize.
-        tokens = term.gsub(%r{[@~"<>{}()+*\-]+}, '* ')
+        tokens = term.gsub(SPECIAL_CHARS, '* +')
         tokens = "#{tokens}*".sub(%r{\* +\*+$}, '*')
```
Comment (lsylvester) [inline lib/winnow/model.rb]: Are we missing a `+` in front of the first term here?

File: lib/winnow/model.rb [hunk 3]
```diff
@@ -83,12 +83,20 @@ def fts_scope_for(column)
...
       end
 
+      def fts_contains_tokens_for(term)
+        term.split(SPECIAL_CHARS)[1..-1].each_with_object([]) do |token, a|
+          a << (token + '*') unless token.empty?
+        end.join(' +')
```
Comment (lsylvester) [inline lib/winnow/model.rb]: Are we missing a `+` in front of the first term here?

File: lib/winnow/model.rb [hunk 4]
```diff
@@ -83,12 +83,20 @@ def fts_scope_for(column)
         "(match(#{table_name}.#{column}) against(? in boolean mode) and (#{table_name}.#{column} like ?))"
       end
 
-      def fts_tokens_for(term)
+      SPECIAL_CHARS = %r{[@~"<>{}()+*\-]+}
```
Comment (lsylvester) [inline lib/winnow/model.rb]: Should white space characters be included in this?

File: lib/winnow/model.rb (hunk 2 above)
Comment (jestingrabbit) [reply lib/winnow/model.rb]: Yeah, I had misread the docs. They were saying no '+*' not no leading plusses entirely.

**Comment by jestingrabbit** [comment]
Having responded to @lsylvester 's suggestions, I'm gonna merge.

**Pull request merged**